packages/core/http/core-http-server/src/router/response_factory.ts

```diff
@@ -27,14 +27,18 @@ export interface KibanaSuccessResponseFactory {
    * Status code: `200`.
    * @param options - {@link HttpResponseOptions} configures HTTP response body & headers.
    */
-  ok(options?: HttpResponseOptions): IKibanaResponse;
+  ok<T extends HttpResponsePayload = HttpResponsePayload>(
```

Comment:
Needed for properly type checking responses. Assuming we only want this for `ok`. But I can easily see this also being checkable for non-OK responses like `400`.